torchdata/dataloader2/reading_service.py

```diff
@@ -285,6 +285,7 @@ def initialize(self, datapipe: DataPipe) -> DataPipe:
                 process_name=f"worker process {worker_id}",
                 call_on_process_init=call_on_process_init,
             )
+            req_queue.cancel_join_thread()
```

Comment:
Emmm, not exactly. This would make sure `req_queue` not being joined by `gc` when the main process wants to exit. Then, it would guarantee that worker process can still access the `TerminationRequest` from `req_queue` rather than any corrupted data from a closed queue